atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManagerTest.java

```diff
@@ -175,4 +177,54 @@ public void registersMetrics() throws InterruptedException {
         assertThat(registry.getTimers().get(SETUP_TASK_METRIC_NAME).getCount()).isGreaterThanOrEqualTo(1);
         assertThat(registry.getTimers().get(FINISH_TASK_METRIC_NAME).getCount()).isGreaterThanOrEqualTo(1);
     }
+
+    @Test
+    public void callsStartTransactionForReadOnlyTransactionsIfFlagIsSet() throws InterruptedException {
+        TimelockService timelockService =
+                spy(new LegacyTimelockService(timestampService, closeableLockService, LockClient.of("lock")));
+        when(closeableLockService.lock(any(), any())).thenReturn(new LockRefreshToken(BigInteger.ONE, Long.MAX_VALUE));
+        SnapshotTransactionManager transactionManager = createSnapshotTransactionManager(timelockService, true);
+
+        transactionManager.runTaskReadOnly(tx -> "ignored");
```

Comment:
Should chat offline, but I believe there are cases where services expect no-op transactions to not have a cost, since some actions are lazy. Should be aware of whether or not we're changing any of those guarantees.

Reply:
you are right, this change _will_ have an effect there; but I think it makes sense to address that in a separate pr after rolling this one out for few specific services.